pkg/cmd/roachtest/tests/restore.go

```diff
@@ -660,6 +660,212 @@ func registerRestore(r registry.Registry) {
 			require.NotEqual(t, 3, maxPauses, "the job should have paused at least once")
 		},
 	})
+
+	for _, sp := range []restoreSpecs{
+		{
+			name:     "restore/nodes=4",
+			hardware: makeHardwareSpecs(hardwareSpecs{}),
+			backup:   makeBackupSpecs(backupSpecs{}),
+			timeout:  2 * time.Hour,
+		},
+		{
+			name:     "restore/gce",
+			hardware: makeHardwareSpecs(hardwareSpecs{cloud: spec.GCE}),
+			backup:   makeBackupSpecs(backupSpecs{}),
+			timeout:  2 * time.Hour,
+		},
+		// TODO(msbutler): add the following tests once roachperf is hooked up and old tests are
```

Comment:
did you consider something like this?
https://github.com/cockroachdb/cockroach/blob/f2e237b0f093d4f4ee8bcef72ac5b699e9451e66/pkg/cmd/roachtest/tests/kv.go#L165-L173
I find it really nice to see each test case in a single line (the down side is that maybe it makes it too easy to add tests!).

Reply:
yeah, i considered this design. What I like about my implementation is that it's very easy to tell what the non-default specs are, as those are the only specs the roachtest writer needs to explicitly specify. For example, the reader can easily see that with `restore/gce`, the specs look identical to `restore/nodes=4` aside from the cloud provider.

Reply:
not sure I got the last thing - how do I know that restore/gce has 4 nodes?

Reply:
take a look at `makeHardwareSpecs()` in restore.go